lightning/src/ln/channel.rs

```diff
@@ -3045,7 +3045,7 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			}
 		}
 
-		let proposed_sat_per_kw = msg.fee_satoshis * 1000 / closing_tx.get_weight() as u64;
+		let proposed_sat_per_kw = msg.fee_satoshis as u32 * 1000 / closing_tx.get_weight() as u32;
```

Comment:
I think this case is technically fine, but I'd feel better if we casted down after doing the arithmetic, not before.